dev-qt/qt-creator/qt-creator-8.0.1.ebuild

```diff
@@ -0,0 +1,440 @@
+"
+
+# qt translations must also be installed or qt-creator translations won't be loaded
+for x in ${PLOCALES}; do
+	IUSE+=" l10n_${x}"
+	RDEPEND+=" l10n_${x}? ( >=dev-qt/qttranslations-${QT_PV} )"
+done
+unset x
+
+# FUNCTION: cmake_use_remove_addsubdirectory
+# USAGE: <flag> <subdir> <files...>
+# DESCRIPTION:
+# <flag> is the name of a flag in IUSE.
+# <subdir> is the  name of a directory called with add_subdirectory().
+# <files...> is a list of one or more qmake project files.
+#
+# This function patches <files> to remove add_subdirectory(<subdir>) from cmake
+# when <flag> is disabled, otherwise it does nothing. This can be useful to
+# avoid an automagic dependency when a subdirectory is added in cmake but the
+# corresponding feature USE flag is disabled. Similar to qt_use_disable_config()
+# from /qt5-build.eclass
+cmake_use_remove_addsubdirectory() {
+	[[ $# -ge 3 ]] || die "${FUNCNAME}() requires at least three arguments"
+	local flag=$1
+	local subdir=$2
+	shift 2
+
+	if ! use "${flag}"; then
+		echo "$@" | xargs sed -i -e "/add_subdirectory(${subdir})/d" || die
+	fi
+}
+
+llvm_check_deps() {
+	has_version -d "sys-devel/clang:${LLVM_SLOT}"
+}
+
+pkg_setup() {
+	if use clang; then
+		llvm_pkg_setup
+		export CLANG_PREFIX="$(get_llvm_prefix ${LLVM_MAX_SLOT})"
+	fi
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	# Remove automagic dep for qt5/qt6
+	sed -e "/^find_package(Qt6/,/else()/ s|if (NOT Qt6_FOUND)|if (1)|" \
+		-i cmake/FindQt5.cmake || die
+
+	# PLUGIN_RECOMMENDS is treated like a hard-dependency
+	sed -i -e '/PLUGIN_RECOMMENDS /d' \
+		src/plugins/*/CMakeLists.txt || die
+
+	cmake_use_remove_addsubdirectory glsl glsl src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory lsp languageserverprotocol \
+		src/libs/CMakeLists.txt tests/auto/CMakeLists.txt
+	cmake_use_remove_addsubdirectory modeling modelinglib \
+		src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory qml advanceddockingsystem \
+		src/libs/CMakeLists.txt
+	cmake_use_remove_addsubdirectory test test \
+		src/plugins/mcusupport/CMakeLists.txt
+
+	# fix translations
+	local languages=()
+	for lang in ${PLOCALES}; do
+		use l10n_${lang} && languages+=( "${lang/-/_}" )
+	done
+	sed -i -e "s|^set(languages.*|set(languages ${languages[*]})|" \
+		share/qtcreator/translations/CMakeLists.txt || die
+
+	# remove bundled yaml-cpp
+	rm -r src/libs/3rdparty/yaml-cpp || die
+
+	# remove bundled qbs
+	rm -r src/shared/qbs || die
+
+	# qt-creator hardcodes the CLANG_INCLUDE_DIR to the default.
+	# However, in sys-devel/clang, the directory changes with respect to
+	# -DCLANG_RESOURCE_DIR.  We sed in the correct include dir.
+	if use clang; then
+		local res_dir="$(${CLANG_PREFIX}/bin/clang -print-resource-dir || die)"
+		sed -i -e "/\w*CLANG_INCLUDE_DIR=/s|=.*|=\"${res_dir}/include\"|" \
+			src/plugins/clangtools/CMakeLists.txt || die
+	fi
+
+	if use doc; then
+		# Fix doc install path
+		sed -i -e "/set(_IDE_DOC_PATH/s|qtcreator|${PF}|" \
```

Comment:
> Are you somehow telling qtcreator where to find them? If not, it may appear to be working because this is a -r0 ebuild so the revision component is omitted from the install path, but it will stop working for -r1 and higher.

Yes,  right here.  I tested the ebuild by temporarily changing `${PF}` to `qt-creator-8.0.1-r2` and can confirm it still shows the documentation in the help viewer.